dask/backends.py

```diff
@@ -104,6 +104,7 @@ def register_inplace(
         self,
         backend: str,
         name: str | None = None,
+        optional: bool = False,
```

Comment:
@ian-r-rose - What do you think about marking creation functions that **should** be covered by compute-based dispatching as `optional`? This way, the backend entrypoint doesn't **need** to override the method, but when the "optional" function is used in `dask.dataframe` or `dask.array`, the corresponding backend will still be registered (and so the underlying cupy -> cudf dispatching rule can be specified in dask_cudf).